src/Service.GraphQLBuilder/Sql/SchemaConverter.cs

```diff
@@ -31,184 +31,387 @@ public static class SchemaConverter
+        /// <param name="entitiesWithManyToManyRelationships">Collection of (source, target) entities which have an M:N relationship between them.</param>
+        /// <param name="foreignKeyFieldsInEntity">Set of fields from source entity holding foreign key references to a target entity.</param>
+        /// <param name="relationshipName">Name of the relationship.</param>
+        /// <param name="relationship">Relationship data.</param>
+        private static FieldDefinitionNode GenerateFieldForRelationship(
+            string entityName,
+            DatabaseObject databaseObject,
+            RuntimeEntities entities,
+            bool isNestedMutationSupported,
+            HashSet<Tuple<string, string>>? entitiesWithManyToManyRelationships,
+            HashSet<string> foreignKeyFieldsInEntity,
+            string relationshipName,
+            EntityRelationship relationship)
+        {
+            // Generate the field that represents the relationship to ObjectType, so you can navigate through it
+            // and walk the graph.
+            SourceDefinition sourceDefinition = databaseObject.SourceDefinition;
+            string targetEntityName = relationship.TargetEntity.Split('.').Last();
+            Entity referencedEntity = entities[targetEntityName];
+            bool isNullableRelationship = false;
+
+            if (// Retrieve all the relationship information for the source entity which is backed by this table definition
+                sourceDefinition.SourceEntityRelationshipMap.TryGetValue(entityName, out RelationshipMetadata? relationshipInfo)
+                &&
+                // From the relationship information, obtain the foreign key definition for the given target entity
+                relationshipInfo.TargetEntityToFkDefinitionMap.TryGetValue(targetEntityName,
+                    out List<ForeignKeyDefinition>? listOfForeignKeys))
+            {
+                ForeignKeyDefinition? foreignKeyInfo = listOfForeignKeys.FirstOrDefault();
+
+                // Determine whether the relationship should be nullable by obtaining the nullability
+                // of the referencing(if source entity is the referencing object in the pair)
+                // or referenced columns (if source entity is the referenced object in the pair).
+                if (foreignKeyInfo is not null)
+                {
+                    RelationShipPair pair = foreignKeyInfo.Pair;
+                    // The given entity may be the referencing or referenced database object in the foreign key
+                    // relationship. To determine this, compare with the entity's database object.
+                    if (pair.ReferencingDbTable.Equals(databaseObject))
+                    {
+                        isNullableRelationship = sourceDefinition.IsAnyColumnNullable(foreignKeyInfo.ReferencingColumns);
+                        foreignKeyFieldsInEntity.UnionWith(foreignKeyInfo.ReferencingColumns);
```

Comment:
by doing union, you account for relationship defined in config and in database?

Reply:
No. This is to track a collection of all the fields which are a referencing column to any entity. Has nothing to do with the config/database scenario. I have added a comment to where this data structure is created and to method summary.